src/model/model.nit

```diff
@@ -627,7 +638,7 @@ class MClassDef
 	var in_hierarchy: nullable POSetElement[MClassDef] = null
 
 	# Is the definition the one that introduced `mclass`?
-	fun is_intro: Bool do return mclass.intro == self
+	fun is_intro: Bool do return isset mclass._intro and mclass.intro == self
```

Comment:
I can use a keyword or better an annotation to avoid reducing the space of
identifiers.
Some suggestions?

~~~
x.a@var = 5
x.a@direct = 5
x.a@attribute = 5
x.a@field = 5
x.var a = 5
x.@a = 5 # ruby syntax!
~~~

... they are all ugly :(



On Tue, Apr 7, 2015 at 9:55 PM, Etienne M. Gagnon <notifications@github.com>
wrote:

> In src/model/model.nit
> <https://github.com/privat/nit/pull/1241#discussion_r27885262>:
>
> > @@ -627,7 +638,7 @@ class MClassDef
> >  	var in_hierarchy: nullable POSetElement[MClassDef] = null
> >
> >  	# Is the definition the one that introduced `mclass`?
> > -	fun is_intro: Bool do return mclass.intro == self
> > +	fun is_intro: Bool do return isset mclass._intro and mclass.intro == self
>
> Suggestions:
>
>    - If one is not meant to have direct low-level access to attributes
>    ever, find another way.
>    - If one in meant to very exceptionally have direct low-level access
>    to attributes, provide a keyword for it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/privat/nit/pull/1241/files#r27885262>.
>